includes/admin/class-wc-stripe-inbox-notes.php

```diff
@@ -0,0 +1,71 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use Automattic\WooCommerce\Admin\Notes\WC_Admin_Note;
+use Automattic\WooCommerce\Admin\Notes\WC_Admin_Notes;
+
+/**
+ * Class that adds inbox notifications.
+ *
+ * @since 4.5.4
+ */
+class WC_Stripe_Inbox_Notes {
+	const NOTE_NAME = 'stripe-apple-pay-domain-verification';
+
+	// TODO Call this only after domain verification has a chance to run
+	public static function notify_of_apple_pay_domain_verification_if_needed() {
+		if ( ! class_exists( 'Automattic\WooCommerce\Admin\Notes\WC_Admin_Notes' ) ) {
+			return;
+		}
+
+		if ( ! class_exists( 'WC_Data_Store' ) ) {
+			return;
+		}
+
+		$stripe_settings       = get_option( 'woocommerce_stripe_settings', array() );
+		$domain_flag_key       = 'apple_pay_domain_set';
+		$verification_complete = isset( $stripe_settings[ $domain_flag_key ] ) && 'yes' === $stripe_settings[ $domain_flag_key ];
+
+		// Only show if in the US.
+		$base_location = wc_get_base_location();
+		if ( ! $base_location || 'US' !== $base_location['country'] ) {
+			return;
+		}
+
+		$data_store = WC_Data_Store::load( 'admin-note' );
+
+		// First, see if we've already created this kind of note so we don't do it again.
+		$note_ids = $data_store->get_notes_with_name( self::NOTE_NAME );
+		if ( ! empty( $note_ids ) ) {
+			$note_id = array_pop( $note_ids );
```

Comment:
You shouldn't need to load the note here - just having a non-empty `$note_ids` array should be enough to return early.

Reply:
The note is loaded primarily so that it can be set to actioned if/when it is effectively complete, in addition to returning early – I borrowed that section from [here](https://github.com/woocommerce/woocommerce-admin/blob/e63eb8f401e14aaf32407523ff664e883cf29eef/src/Notes/WooCommercePayments.php#L54-L71) – does that seem right, or could it be done a different way?

Reply:
That seems right to me